digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadAddressService.scala

```diff
@@ -132,25 +144,27 @@ class RoadAddressService(roadLinkService: RoadLinkService, eventbus: DigiroadEve
     //TODO use complementedIds instead of only roadLinkIds below. There is no complementary ids for changeInfo dealing (for now)
     val combinedFuture=
       for {
-        changedRoadLinksF<- Future(roadLinkService.getChangeInfoFromVVH(boundingRectangle, municipalities))
-        complementaryFuture <- Future (fetchRoadLinksWithComplementary(boundingRectangle, roadNumberLimits, municipalities, everything, publicRoads))
-        fetchRoadAddressesByBoundingBoxF <- Future(fetchRoadAddressesByBoundingBox(boundingRectangle))
-      }  yield( changedRoadLinksF, complementaryFuture,fetchRoadAddressesByBoundingBoxF)
-    val (changedRoadLinks,((roadLinks, complementaryLinks)),(floatingViiteRoadLinks, addresses, floating)) =
+        changedRoadLinksF <- boundingBoxResult.changeInfoF
+        roadLinkFuture <- boundingBoxResult.roadLinkF
+        complementaryFuture <- boundingBoxResult.complementaryF
+        fetchRoadAddressesByBoundingBoxF <- boundingBoxResult.roadAddressResultF
+      }  yield( changedRoadLinksF, (roadLinkFuture, complementaryFuture),fetchRoadAddressesByBoundingBoxF)
+    val (changedRoadLinks,((roadLinks, complementaryLinks)),roadAddressResults) =
       withTiming(
         Await.result(combinedFuture, Duration.Inf), "End fetch vvh road links in %.3f sec"
       )
+    val (floatingViiteRoadLinks, addresses, floating) = (roadAddressResults.historyLinkAddresses, roadAddressResults.current, roadAddressResults.floating)
```

Comment:
floatingViiteRoadLinks are actually historyLinkAddresses

Reply:
Should rename